main.py

```diff
@@ -0,0 +1,46 @@
+#!/usr/bin/env python3
+
+global rank
+
+class Point:
+    def __init__(self, i, p):
+        self.i = i
+        self.x, self.y = p
+
+def findRank(p):
+    if len(p) <= 1:
+        return p
+    i, li, ri = 0, 0, 0
+    l = findRank(p[:int(len(p)/2)])
+    r = findRank(p[int(len(p)/2):])
+    while True:
+        if l[li].y <= r[ri].y:
+            p[i] = l[li]
+            li += 1
+            if li >= len(l):
+                for n in r[ri:]:
```

Comment:
The name `n` is confused with the input `n`, which denotes the number of points.